### PR TITLE
Remove legacy plotting

### LINKED ISSUE #2879: Remove old plotting backend from Scipp (August 2023)
Removal is scheduled for v23.08.0 or after.

- [ ] Remove code (`scipp.plotting`)
- [ ] Remove tests. Consider keeping a minimal test suite for the plot method.
- [ ] Remove migration guide after the backend was removed.
- [ ] Consider adding forwards or special pages explaining the change to replace the current plotting details pages, in case people still use those links.

### PR DESCRIPTION
**Pull request by nvaytet**
Does what it says on the tin!

Fix #2879

### CODE REVIEW

File: docs/visualization/plotting.ipynb [hunk 1]
```diff
@@ -1,31 +1,17 @@
...
-    "# Plotting Overview\n",
+    "# Plotting\n",
     "\n",
     "## Getting started\n",
     "\n",
-    "Scipp offers a number of different ways to plot data from a `Variable`, `DataArray`, or a `Dataset`.\n",
-    "It uses the `matplotlib` graphing library to do so, as well as the `pythreejs` project for 3D visualizations.\n",
+    "Scipp offers a number of different ways to plot data from a `Variable`, `DataArray`, `Dataset` or a `DataGroup`.\n",
+    "\n",
+    "It uses the `plopp` package to do so (see https://scipp.github.io/plopp).\n",
```
Comment (SimonHeybrock) [inline docs/visualization/plotting.ipynb:14]: Shouldn't we begin this docs page with install instructions for plopp?

File: lib/python/operations.cpp [hunk 2]
```diff
@@ -93,21 +93,6 @@ void init_operations(py::module &m) {
...
-      },
-      py::call_guard<py::gil_scoped_release>());
-  m.def("get_slice_params", [](const Variable &var, const Variable &coord,
-                               const Variable &begin, const Variable &end) {
-    const auto [dim, start, stop] =
-        get_slice_params(var.dims(), coord, begin, end);
-    // Do NOT release GIL since using py::slice
-    return std::tuple{dim.name(), py::slice(start, stop, 1)};
-  });
-
```
Comment (SimonHeybrock) [inline lib/python/operations.cpp:110]: Any reason for removing this? Could be useful in the future?

File: src/scipp/configuration/config_default.yaml [hunk 3]
```diff
@@ -26,109 +26,3 @@ colors:
 
 # Maximum number of rows in a table to display at the same time.
 table_max_size: 50
-
-# Default style of plots produced by scipp.
```
Comment (SimonHeybrock) [inline src/scipp/configuration/config_default.yaml:30]: Is this found in Plopp, or are the settings no longer required?

File: tools/manual_testing/plot.ipynb [hunk 4]
```diff
@@ -1,602 +0,0 @@
-{
- "cells": [
-  {
-   "cell_type": "markdown",
-   "id": "1f8cc565-023d-4ee5-b0fa-2c4ee6aa2483",
-   "metadata": {},
-   "source": [
-    "# Overview: Manual testing for plotting functionality\n",
```
Comment (SimonHeybrock) [inline tools/manual_testing/plot.ipynb:8]: Is it worth moving this notebook to plopp?

**Review by SimonHeybrock — COMMENTED** [review]
- The is no test of the `plot()` method, as far as I can tell (both that it works with plopp, and that it gives a reasonable error without plopp). Is it worth adding such an env?
- `conda/meta.yaml` needs updating, in particular the test requirements.

File: src/scipp/configuration/config_default.yaml (hunk 3 above)
Comment (nvaytet) [reply src/scipp/configuration/config_default.yaml:30]: There is no config in Plopp currently.
We may add one again in the future, if users want one.

File: lib/python/operations.cpp (hunk 2 above)
Comment (nvaytet) [reply lib/python/operations.cpp:110]: I was following the instructions in https://github.com/scipp/scipp/blob/main/src/scipp/core/__init__.py#L46  ;-)
I don't know if we want to keep it or not.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**